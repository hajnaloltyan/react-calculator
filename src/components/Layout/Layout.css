.header {
  display: flex;
  justify-content: space-between;
  margin: 2% 5%;
}

.title {
  font-size: 40px;
  font-weight: 800;
}
