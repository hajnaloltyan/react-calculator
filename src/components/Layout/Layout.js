import React from 'react';
import { Outlet } from 'react-router-dom';
import Navbar from '../Navbar/Navbar';
import './Layout.css';

const Layout = () => (
  <>
    <header className="header">
      <h1 className="title">Math Magicians</h1>
      <Navbar />
    </header>

    <main className="content">
      <Outlet />
    </main>
  </>
);

export default Layout;
