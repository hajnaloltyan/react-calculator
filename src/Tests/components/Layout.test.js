import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Layout from '../../components/Layout/Layout';

describe('Test the Layout component', () => {
  test('Check if the Layout component render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/']}>
        <Layout />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Check that all the correct elements are in the component', () => {
    const component = render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>,
    );

    const title = component.container.querySelector('h1');

    expect(title).toHaveTextContent('Math Magicians');
  });
});
