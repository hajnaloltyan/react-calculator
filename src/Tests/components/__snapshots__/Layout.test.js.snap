// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Layout component Check if the Layout component render correctly 1`] = `
Array [
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Math Magicians
    </h1>
    <nav
      className="navbar"
    >
      <ul
        className="navlinks"
      >
        <li>
          <a
            aria-current="page"
            className="active active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="active"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            className="active"
            href="/quotes"
            onClick={[Function]}
          >
            Quotes
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="content"
  />,
]
`;
